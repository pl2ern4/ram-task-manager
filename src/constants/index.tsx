const USER = ['Alina Jacob', 'Daniel Bens', 'Jina Thomas','Not Assigned'];

const TICKET_TYPE = ['Story', 'Issue', 'Maintaince','Not Assigned'];

const TICKET_STATUS = ['Todo', 'inProgress', 'Done',];

const TICKET_ORDERBY = {
    modifiedDate: 'MODIFIED_DATE',
    createdDate: 'CREATED_DATE',
    bookmarked: 'BOOKMARKED',
    title: 'TITLE'
};

const ALERT_CONSTANT = {
    story: 'success',
    bug: 'error',
    maintaince:'info'
  }

export {USER, TICKET_TYPE, TICKET_STATUS, TICKET_ORDERBY, ALERT_CONSTANT}