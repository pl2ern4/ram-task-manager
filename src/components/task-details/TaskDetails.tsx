import BookmarkIcon from '@mui/icons-material/Bookmark';
import Button from '@mui/material/Button';
import { styled } from '@mui/material/styles';
import DatePicker from "react-datepicker";
import "react-datepicker/dist/react-datepicker.css";
import SelectInput from '../select-input/SelectInput';
import EditableInput from '../editable-label/EditableInput';
import Item from '../item/Item';
import { USER, TICKET_TYPE, TICKET_STATUS } from '../../constants';
import { TaskBookmarked } from '../card/styles';


const TaskDetailWrapper = styled(Item)`
    width: fit-content;
    margin: 1rem;
`;

const TaskOtherDetails = styled('div')`
    display: flex;
    &>label{
        flex:.5;
    }
    >&div{
        flex: 2;
    }
`;

type Props = {
    status: string;
    asignee?: string;
    taskType?: string;
    storyPoint?: number;
    deadLine: Date| string |null;
    isBookMarked?: boolean;
    handleChange: (fieldName:string, value:any)=>void;
    deleteTicket: ()=>void;
}

function TaskDetails({ deleteTicket, deadLine = null, status = 'Todo', asignee = 'Not Assigned', taskType = 'Not Assigned', storyPoint, handleChange, isBookMarked }: Props) {
    return (
        <TaskDetailWrapper>
            <TaskOtherDetails>
                <label>Favorite: </label>
                <TaskBookmarked onClick={()=>handleChange("isBookMarked",!isBookMarked)}><BookmarkIcon sx={{color:isBookMarked? '#e68a12': '#808080b8'}} /></TaskBookmarked>
            </TaskOtherDetails>
            <SelectInput label='status' defaultOption={status} options={TICKET_STATUS} onChange={(value:any)=>handleChange("status",value?.target?.textContent)}/>
            <SelectInput label='Asignee' defaultOption={asignee} options={USER}  onChange={(value:any)=>handleChange("assignee",value?.target?.textContent)}/>
            <SelectInput label='Task Type' defaultOption={taskType} options={TICKET_TYPE} onChange={(value:any)=>handleChange("taskType",value?.target?.textContent)}/>
            <EditableInput label='Story Points' text={storyPoint} type="number" onChange={(value:any)=>handleChange("storyPoint",value)}/>
            <TaskOtherDetails>
                <label>Task DeadLine</label>
                <DatePicker selected={deadLine} onChange={(value:any)=>handleChange("deadlineDate",value)}/>
            </TaskOtherDetails>
            <TaskOtherDetails>
                <Button variant="outlined" sx={{color:'red', margin: '1rem'}} onClick={()=>deleteTicket()}>Delete This ticket?</Button>
            </TaskOtherDetails>
        </TaskDetailWrapper>

    );
}
export default TaskDetails;