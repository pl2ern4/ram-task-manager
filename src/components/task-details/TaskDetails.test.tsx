import * as React from 'react';
import { render } from '@testing-library/react';
import TaskDetails from './TaskDetails';
import { USER, TICKET_TYPE, TICKET_STATUS } from '../../constants';

describe('SelectInput', () => {
    it('should Match Snapshot', async () => {
        const { container } = render(<TaskDetails handleChange={jest.fn()} asignee={USER[3]} status={TICKET_STATUS[1]} taskType={TICKET_TYPE[1]}/>)
        expect(container).toMatchSnapshot();
    });  
    it('should Match Snapshot, when asignee, taskType not passed', async () => {
        const { container } = render(<TaskDetails handleChange={jest.fn()} status={TICKET_STATUS[0]} />)
        expect(container).toMatchSnapshot();
    });    
})