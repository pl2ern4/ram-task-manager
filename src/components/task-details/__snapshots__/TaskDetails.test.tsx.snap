// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectInput should Match Snapshot 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-oy7y5j-MuiPaper-root"
  >
    <div
      class="css-3pg3o7"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-s86xg-MuiSvgIcon-root"
        data-testid="BookmarkIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M17 3H7c-1.1 0-1.99.9-1.99 2L5 21l7-3 7 3V5c0-1.1-.9-2-2-2"
        />
      </svg>
    </div>
    <div
      class="css-1dw64bn"
    >
      <label
        class="css-14pxzl7"
      >
        status
      </label>
      <button
        aria-controls=":r0:"
        aria-expanded="false"
        class="base-Select-root css-12xgkxt"
        role="combobox"
        type="button"
      >
        inProgress
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="UnfoldMoreRoundedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="m12 5.83 2.46 2.46c.39.39 1.02.39 1.41 0 .39-.39.39-1.02 0-1.41L12.7 3.7a.9959.9959 0 0 0-1.41 0L8.12 6.88c-.39.39-.39 1.02 0 1.41.39.39 1.02.39 1.41 0zm0 12.34-2.46-2.46a.9959.9959 0 0 0-1.41 0c-.39.39-.39 1.02 0 1.41l3.17 3.18c.39.39 1.02.39 1.41 0l3.17-3.17c.39-.39.39-1.02 0-1.41a.9959.9959 0 0 0-1.41 0z"
          />
        </svg>
      </button>
      <input
        aria-hidden="true"
        style="border: 0px; height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; white-space: nowrap; width: 1px;"
        tabindex="-1"
        value="inProgress"
      />
    </div>
    <div
      class="css-1dw64bn"
    >
      <label
        class="css-14pxzl7"
      >
        Asignee
      </label>
      <button
        aria-controls=":r1:"
        aria-expanded="false"
        class="base-Select-root css-12xgkxt"
        role="combobox"
        type="button"
      >
        Not Assigned
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="UnfoldMoreRoundedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="m12 5.83 2.46 2.46c.39.39 1.02.39 1.41 0 .39-.39.39-1.02 0-1.41L12.7 3.7a.9959.9959 0 0 0-1.41 0L8.12 6.88c-.39.39-.39 1.02 0 1.41.39.39 1.02.39 1.41 0zm0 12.34-2.46-2.46a.9959.9959 0 0 0-1.41 0c-.39.39-.39 1.02 0 1.41l3.17 3.18c.39.39 1.02.39 1.41 0l3.17-3.17c.39-.39.39-1.02 0-1.41a.9959.9959 0 0 0-1.41 0z"
          />
        </svg>
      </button>
      <input
        aria-hidden="true"
        style="border: 0px; height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; white-space: nowrap; width: 1px;"
        tabindex="-1"
        value="Not Assigned"
      />
    </div>
    <div
      class="css-1dw64bn"
    >
      <label
        class="css-14pxzl7"
      >
        Task Type
      </label>
      <button
        aria-controls=":r2:"
        aria-expanded="false"
        class="base-Select-root css-12xgkxt"
        role="combobox"
        type="button"
      >
        Issue
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="UnfoldMoreRoundedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="m12 5.83 2.46 2.46c.39.39 1.02.39 1.41 0 .39-.39.39-1.02 0-1.41L12.7 3.7a.9959.9959 0 0 0-1.41 0L8.12 6.88c-.39.39-.39 1.02 0 1.41.39.39 1.02.39 1.41 0zm0 12.34-2.46-2.46a.9959.9959 0 0 0-1.41 0c-.39.39-.39 1.02 0 1.41l3.17 3.18c.39.39 1.02.39 1.41 0l3.17-3.17c.39-.39.39-1.02 0-1.41a.9959.9959 0 0 0-1.41 0z"
          />
        </svg>
      </button>
      <input
        aria-hidden="true"
        style="border: 0px; height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; white-space: nowrap; width: 1px;"
        tabindex="-1"
        value="Issue"
      />
    </div>
    <div
      class="css-psohlv"
    >
      <label
        class="css-eie7o"
      >
        Story Points
      </label>
      <label
        class="css-17045cg"
      />
    </div>
  </div>
</div>
`;

exports[`SelectInput should Match Snapshot, when asignee, taskType not passed 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-oy7y5j-MuiPaper-root"
  >
    <div
      class="css-3pg3o7"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-s86xg-MuiSvgIcon-root"
        data-testid="BookmarkIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M17 3H7c-1.1 0-1.99.9-1.99 2L5 21l7-3 7 3V5c0-1.1-.9-2-2-2"
        />
      </svg>
    </div>
    <div
      class="css-1dw64bn"
    >
      <label
        class="css-14pxzl7"
      >
        status
      </label>
      <button
        aria-controls=":re:"
        aria-expanded="false"
        class="base-Select-root css-12xgkxt"
        role="combobox"
        type="button"
      >
        Todo
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="UnfoldMoreRoundedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="m12 5.83 2.46 2.46c.39.39 1.02.39 1.41 0 .39-.39.39-1.02 0-1.41L12.7 3.7a.9959.9959 0 0 0-1.41 0L8.12 6.88c-.39.39-.39 1.02 0 1.41.39.39 1.02.39 1.41 0zm0 12.34-2.46-2.46a.9959.9959 0 0 0-1.41 0c-.39.39-.39 1.02 0 1.41l3.17 3.18c.39.39 1.02.39 1.41 0l3.17-3.17c.39-.39.39-1.02 0-1.41a.9959.9959 0 0 0-1.41 0z"
          />
        </svg>
      </button>
      <input
        aria-hidden="true"
        style="border: 0px; height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; white-space: nowrap; width: 1px;"
        tabindex="-1"
        value="Todo"
      />
    </div>
    <div
      class="css-1dw64bn"
    >
      <label
        class="css-14pxzl7"
      >
        Asignee
      </label>
      <button
        aria-controls=":rf:"
        aria-expanded="false"
        class="base-Select-root css-12xgkxt"
        role="combobox"
        type="button"
      >
        Not Assigned
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="UnfoldMoreRoundedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="m12 5.83 2.46 2.46c.39.39 1.02.39 1.41 0 .39-.39.39-1.02 0-1.41L12.7 3.7a.9959.9959 0 0 0-1.41 0L8.12 6.88c-.39.39-.39 1.02 0 1.41.39.39 1.02.39 1.41 0zm0 12.34-2.46-2.46a.9959.9959 0 0 0-1.41 0c-.39.39-.39 1.02 0 1.41l3.17 3.18c.39.39 1.02.39 1.41 0l3.17-3.17c.39-.39.39-1.02 0-1.41a.9959.9959 0 0 0-1.41 0z"
          />
        </svg>
      </button>
      <input
        aria-hidden="true"
        style="border: 0px; height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; white-space: nowrap; width: 1px;"
        tabindex="-1"
        value="Not Assigned"
      />
    </div>
    <div
      class="css-1dw64bn"
    >
      <label
        class="css-14pxzl7"
      >
        Task Type
      </label>
      <button
        aria-controls=":rg:"
        aria-expanded="false"
        class="base-Select-root css-12xgkxt"
        role="combobox"
        type="button"
      >
        Not Assigned
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="UnfoldMoreRoundedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="m12 5.83 2.46 2.46c.39.39 1.02.39 1.41 0 .39-.39.39-1.02 0-1.41L12.7 3.7a.9959.9959 0 0 0-1.41 0L8.12 6.88c-.39.39-.39 1.02 0 1.41.39.39 1.02.39 1.41 0zm0 12.34-2.46-2.46a.9959.9959 0 0 0-1.41 0c-.39.39-.39 1.02 0 1.41l3.17 3.18c.39.39 1.02.39 1.41 0l3.17-3.17c.39-.39.39-1.02 0-1.41a.9959.9959 0 0 0-1.41 0z"
          />
        </svg>
      </button>
      <input
        aria-hidden="true"
        style="border: 0px; height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; white-space: nowrap; width: 1px;"
        tabindex="-1"
        value="Not Assigned"
      />
    </div>
    <div
      class="css-psohlv"
    >
      <label
        class="css-eie7o"
      >
        Story Points
      </label>
      <label
        class="css-17045cg"
      />
    </div>
  </div>
</div>
`;
